plugins/social_network/social_network.cpp

```diff
@@ -89,6 +171,222 @@ namespace golos { namespace plugins { namespace social_network {
+                    }
+                    
+                    if ((!depth_parameters.has_comment_body_depth || depth_parameters.comment_body_depth > 0) && o.body.size() < 1024*1024*128) {
+                        from_string(con.body, o.body);
+                    }
+
+                    if ((!depth_parameters.has_comment_json_metadata_depth || depth_parameters.comment_json_metadata_depth > 0) &&
+                        fc::is_utf8(o.json_metadata)
+                    ) {
+                        from_string(con.json_metadata, o.json_metadata);
+                    }
+
+                    con.block_number = db.head_block_num();
+                });
+            }
+        }
+
+        // Checking should we delete comment_content if needed depth has been expired
+        void operator()(const comment_payout_update_operation& o) const { 
+            if (!depth_parameters.need_clear()) {
+                return;
+            }
+
+            const auto& comment = db.get_comment(o.author, o.permlink);
+
+            auto* content_ptr = find_comment_content(comment.id);
+            if (content_ptr == nullptr) {
+                return;
+            }
+            auto& content = *content_ptr;
+
+
+            auto delta = db.head_block_num() - content.block_number;
+
+            if (depth_parameters.should_delete_whole_content_object(delta)) {
+                db.remove(content);
+                return;
+            }
+
+            if (depth_parameters.should_delete_part_of_content_object(delta)) {
+                db.modify(content, [&](comment_content_object& con) {
+                    if (delta > depth_parameters.comment_title_depth) {
+                        con.title.clear();
+                    }
+
+                    if (delta > depth_parameters.comment_body_depth) {
+                        con.body.clear();
+                    }
+
+                    if (delta > depth_parameters.comment_json_metadata_depth) {
+                        con.json_metadata.clear();
+                    }
+                });
+
+                return; 
+            }
+        }
+
+    };
+
+    void social_network::impl::post_operation(const operation_notification &o) {
+        try {
+            auto& db = database();
+
+            operation_visitor ovisit(db, depth_parameters);
+
+            o.op.visit(ovisit);
+        } FC_CAPTURE_AND_RETHROW()
+    }
+    void social_network::impl::on_block(const signed_block &b) {
+        try {
+            auto & db = database();
+
+            const auto &idx = db.get_index<comment_content_index>().indices().get<by_block_number>();
+
+            for (auto itr = idx.begin(); itr != idx.end();) {
+                auto & content = *itr;
+                ++itr;
+
+                const auto &cidx = db.get_index<comment_index>().indices().get<by_id>();
+            
+                auto comment = cidx.find(content.comment);
+
+                int64_t cash_window_sec = STEEMIT_CASHOUT_WINDOW_SECONDS;
```

Comment:
it's better to use direct instead if indirect properties. there were two payout windows earlier who knows what will be in the future : )
Please, use `comment.mode` to check it comment archived. In addition, it will remove `config.hpp` dependency.